packages/vite/src/node/plugins/dynamicImportVars.ts

```diff
@@ -39,14 +39,31 @@ interface DynamicImportPattern {
   rawPattern: string
 }
 
-const dynamicImportHelper = (glob: Record<string, any>, path: string) => {
+const dynamicImportHelper = (
+  glob: Record<string, any>,
+  path: string,
+  rawPath: string,
+) => {
   const v = glob[path]
   if (v) {
     return typeof v === 'function' ? v() : Promise.resolve(v)
   }
+  let _path = path
+  rawPath.split(/\$\{.+\}/).forEach((str) => {
+    _path = _path.replace(str, '')
+  })
```

Comment:
I don't quite understand this check, isn't `rawPath` something like `./foo/${var}.js`. And `_path` would be `var` and rarely includes a `/`?

I was thinking something like:

```js
`__variableDynamicImportRuntimeHelper(${glob}, \`${rawPattern}\`, ${rawPattern.split('/').length})`

...
const dynamicImportHelper = (
   glob: Record<string, any>,
   path: string,
   segs: number,
 ) => {

...

path.split('/').length !== segs ? '. Note that variables...' : ''
```

Reply:
Thanks for the tip, I seem to be thinking in the wrong direction.